### PR TITLE
[WIP] MAISTRA-2316 Automatic exchange of trust bundles 2316

### PR DESCRIPTION
**Pull request by dgn**

### CODE REVIEW

**Comment by dgn** [comment]
This works in unit tests but I have yet to confirm it does in a federation. Will run some real-world tests tomorrow

File: pkg/servicemesh/federation/server/trustbundle.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+
+	"github.com/lestrrat-go/jwx/jwk"
+)
+
+type bundleDoc struct {
```
Comment (shamsher31) [inline pkg/servicemesh/federation/server/trustbundle.go:30]: nit:
```suggestion
type trustBundle struct {
```

File: pkg/servicemesh/federation/server/trustbundle.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	RefreshHint int       `json:"spiffe_refresh_hint,omitempty"`
+}
+
+func (s *Server) handleTrustBundle(response http.ResponseWriter, request *http.Request) {
+	s.RLock()
+	defer s.RUnlock()
+	respBytes, err := json.Marshal(s.trustBundle)
+	if err != nil {
+		s.logger.Errorf("failed to marshal to json: %s", err)
+		response.WriteHeader(500)
```
Comment (shamsher31) [inline pkg/servicemesh/federation/server/trustbundle.go:42]: nit:
```suggestion
		response.WriteHeader(http.StatusInternalServerError)
```

File: pkg/servicemesh/federation/server/trustbundle.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		return
+	}
+	_, err = response.Write(respBytes)
+	if err != nil {
+		s.logger.Errorf("failed to send response: %s", err)
+		response.WriteHeader(500)
```
Comment (shamsher31) [inline pkg/servicemesh/federation/server/trustbundle.go:48]: nit:
```suggestion
		response.WriteHeader(http.StatusInternalServerError)
```

**Comment by jwendell** [comment]
@dgn What's the status of this? Can it be safely closed?

**Pull request closed without merging**